woocommerce/woocommerce
# Perform feature flag change and redirection in the same request

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38832



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note: Before testing this make sure `Try the new product editor (Beta)` under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and `new-product-management-experience` under `/wp-admin/tools.php?page=woocommerce-admin-test-helper` are both turned off.

1. Make sure you add the experiment `woocommerce_product_creation_experience_202306_v2` and pick the variation you want to test like follow. 

https://github.com/woocommerce/woocommerce/assets/13334210/5baa9945-6906-428c-ab3c-65e1712695cc

2. Go to `/wp-admin/admin.php?page=wc-admin&task=products`
3. Then click on `Physical product` item.
4. If the variation name chosen was `control` then you should see the old/legacy product form
5. If the variation was `treatment` then you should see the new product blocks editor and anytime from now you should not longer see the old one anymore unless you disable the `Try the new product editor (Beta)` feature.
